src/foundry/foundry.js/clientDocuments/actor.d.ts

```diff
@@ -29,6 +31,75 @@ declare global {
      * A convenient reference to the file path of the Actor's profile image
      */
     get img(): this['data']['img'];
+
+    /**
+     * Provide an object which organizes all embedded Item instances by their type
+     */
+    get itemTypes(): Record<string, Item[]>;
+    /**
```

Comment:
The line between members is missing.

Reply:
Shouldn't `ItemTypes` return a record containing the configured Item type instead of a generic item?

Reply:
> Shouldn't `ItemTypes` return a record containing the configured Item type instead of a generic item?

There is nothing one can pull to resonantly infer a better typing. At best, we could try to drag it in via generics and make it default to this. Probably best addressed after pinning down the generics for the system data for both Item and Actor classes.

Reply:
Yes, there is: Use `ConfiguredDocumentClass` and `InstanceType`. Basically, there is a new way for the user to configure the types for the document classes, which we introduced in the 0.8.c branch (i.e. it’s totally different on 0.7.x). Takes a look at config.d.ts, it contains some examples that show how it works. No generics needed for all of that on the user side.